pysb/simulator/base.py

```diff
@@ -108,29 +116,61 @@ def initials(self, new_initials):
                                                  'MonomerPattern or '
                                                  'ComplexPattern' %
                                                  repr(cplx_pat))
+                    # if val is a number, convert it to a single-element array
+                    if not isinstance(val, collections.Iterable):
```

Comment:
I think this should be `Sequence` instead of `Iterable`. You need `Sequence` if you want to retrieve an element from a specific index, which I believe the code using these values will do. I see this mistake in a bunch of other places. I'll try to make a note on the others.